### DISCUSSION TITLE
Few quetions about request files: file-parameters-with-uploadfile

### DISCUSSION THREAD

**Discussion by LuchiLucs** [fastapi/fastapi#12714] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class EnumFileCategory (str, Enum):
    A = "A"
    B = "B"

@router.post("/upload-files/")
async def upload_files(
    files: list[UploadFile] = File(...),
    category: EnumFileCategory = Form(...),
):
    for file in files:
        try:
            contents = await file.read()
            await do_work_example()
        finally:
            await file.close()
    return {
        "category": category,
        "filenames": [f.filename for f in files],
    }
```

### Description

1. The documentation [here](https://fastapi.tiangolo.com/tutorial/request-files/#file-parameters-with-uploadfile) says to install the python package `python-multipart` but after coding an endpoint as follows, I succesfully tested it using the _/docs_ without problems. Am I missing something?
2. I tested it with `.pdf` files, while with `.md` files I had problems. Am I missing something? Is it related to "binary" vs "textual" files?

### Operating System

Linux

### FastAPI Version

0.115.0

### Pydantic Version

3.9.2

### Python Version

4.12.6

**Comment by luzzodev** [comment, marked as answer]
1) `python-multipart` is required to be installed if you install Fastapi without external dependencies, so if you have installed:

- `pip install fastapi[standard]`
- `pip install fastapi[all]`
It comes with external dependencies as a required one. `python-multipart >=0.0.7`

2 Can you explain what your issue is? What is the expected result and the unexpected one?

**Reply by LuchiLucs** [reply]
1. Ok thank you. I installed it with `[standard]` so I guess it was included.
2. I will come back as soon as possible, thank you.
